storage/kubernetes/controller.go

```diff
@@ -119,27 +120,31 @@ func (s *storage) init(secrets v1controller.SecretController) {
 }
 
 func (s *storage) Get() (*v1.Secret, error) {
-	s.Lock()
-	defer s.Unlock()
+	s.RLock()
+	defer s.RUnlock()
 
 	return s.storage.Get()
 }
 
 func (s *storage) targetSecret() (*v1.Secret, error) {
+	s.RLock()
+	defer s.RUnlock()
+
 	existingSecret, err := s.secrets.Get(s.namespace, s.name, metav1.GetOptions{})
 	if errors.IsNotFound(err) {
 		return &v1.Secret{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      s.name,
 				Namespace: s.namespace,
 			},
+			Type: v1.SecretTypeTLS,
```

Comment:
Is this a breaking change? I assume that the secrets created with this previously we `Opaque` secrets because this is missing. Which means that the `cert.IsValidTLSSecret` may fail when it isn't supposed to for upgraded instances?

Reply:
I did pull the type check from the helper, just in case there were some certs stored in the `file` cache, which wouldn't set the type if the Secret wasn't ever round-tripped through Kubernetes or the merge operation.